ENIGMAsystem/SHELL/Universal_System/Extensions/VideoPlayer/videoplayer.cpp

```diff
@@ -0,0 +1,329 @@
+}
+
+void splash_set_stop_mouse(bool stop) {
+  splash_get_stop_mouse = stop;
+}
+
+void splash_set_volume(int vol) {
+  splash_get_volume = vol;
+}
+
+void splash_show_video(string fname, bool loop, string window_id) {
+  string video, wid, wstr, hstr, xstr, ystr, 
+    size, pstn, geom, flls, brdr, ontp, lpng;
+
+  if (splash_get_main) { // embeds inside game window
+    #ifdef __APPLE__
+      #ifdef __MACH__
+        // this is not used by my extension
+        #ifndef VIDEO_PLAYER_SELF_CONTAINED
+          wid = cocoa_window_get_contentview(wid.c_str());
+        #else
+          wid = cocoa_window_get_contentview(window_id.c_str());
+        #endif
+      #endif
+    #else
+      // this is not used by my extension
+      #ifndef VIDEO_PLAYER_SELF_CONTAINED  
+        wid = window_identifier();
+      #else
+        wid = window_id;
+      #endif
+    #endif
+  } else { 
+    // creates a new window to play splash
+    // -1 is guaranteed to be not a window
+    wid = "-1";
+  }
+
+  splash_get_interupt = !splash_get_main ? splash_get_interupt : true;
+  flls = !splash_get_main ? (splash_get_fullscreen ? "yes" : "no") : "no";
+  brdr = !splash_get_main ? (splash_get_border ? "yes" : "no") : "no";
+  ontp = !splash_get_main ? (splash_get_top ? "yes" : "no") : "no";
+  lpng = loop ? "yes" : "no";
+
+  wstr = std::to_string(splash_get_width);
+  hstr = std::to_string(splash_get_height);
+  xstr = std::to_string(splash_get_x);
+  ystr = std::to_string(splash_get_y);
+  size = wstr + "x" + hstr;
+  if (splash_get_x == INT_MAX || splash_get_y == INT_MAX) {
+    pstn = "50%+50%";
+    geom = size + "+" + pstn;
+  } else {
+    pstn = xstr + "+" + ystr;
+    geom = size + "+" + pstn;
+  }
+
+  if (video_exists(video)) {
+    video_delete(video);
+  }
+ 
+  video = video_add(fname);
+
+  std::ofstream file;
+  file.open ("input.conf");
+  file << "CLOSE_WIN ignore\n";
+  if (splash_get_stop_mouse) {
+    file << "MBTN_LEFT quit\n";
+    file << "MBTN_RIGHT quit\n";
+  }
+  file.close();
+
+  video_set_option_string(video, "input-conf", "input.conf");
+  video_set_option_string(video, "volume", std::to_string(splash_get_volume));
+  video_set_option_string(video, "input-default-bindings", "no");
+  video_set_option_string(video, "title", splash_get_caption);
+  video_set_option_string(video, "fs", flls);
+  video_set_option_string(video, "border", brdr);
+  video_set_option_string(video, "keepaspect-window", "no");
+  video_set_option_string(video, "taskbar-progress", "no");
+  video_set_option_string(video, "ontop", ontp);
+  video_set_option_string(video, "geometry", geom);
+  video_set_option_string(video, "config", "no");
+  video_set_option_string(video, "loop", lpng);
+  video_set_option_string(video, "osc", "no");
+  video_set_option_string(video, "wid", wid);
+
+  video_play(video);
+  if (splash_get_interupt) {
+    while (video_is_playing(video)) {
+      std::this_thread::sleep_for(std::chrono::milliseconds(5));
+      #ifdef _WIN32
```

Comment:
@fundies i know how much you hate these platform macros so please don't worry I already intend on removing them, I wanted to tell you in advance I'm going to separate this into a windows and mac helper and a non-windows/mac function stub in the Makefile because it'll be a while before i can do it. I need to discuss with Josh something first in regards to the OS X (posix) equivalent to Win32 PeekMessage. Linux doesn't need the posix equivalent to peekmessage for this code to work but windows and mac will have the program not respond and crash without a message pump of some kind.